gossip/src/cluster_info.rs

```diff
@@ -2762,49 +2761,74 @@ impl Node {
     ) -> Self {
         let (gossip_port, (gossip, ip_echo)) =
             Self::get_gossip_port(gossip_addr, port_range, bind_ip_addr);
-        let (tvu_port, tvu) = Self::bind(bind_ip_addr, port_range);
-        let (tvu_quic_port, tvu_quic) = Self::bind(bind_ip_addr, port_range);
-        let udp_config = SocketConfig { reuseport: false };
-        let quic_config = SocketConfig { reuseport: true };
+
+        let read_write_socket_config = SocketConfig::default();
+        let (tvu_port, tvu) =
+            Self::bind_with_config(bind_ip_addr, port_range, read_write_socket_config.clone());
+        let (tvu_quic_port, tvu_quic) =
+            Self::bind_with_config(bind_ip_addr, port_range, read_write_socket_config.clone());
+        let tpu_udp_config = SocketConfig::default();
+        let tpu_quic_config = SocketConfig::default().reuseport(true);
         let ((tpu_port, tpu), (_tpu_quic_port, tpu_quic)) =
             bind_two_in_range_with_offset_and_config(
                 bind_ip_addr,
                 port_range,
                 QUIC_PORT_OFFSET,
-                udp_config.clone(),
-                quic_config.clone(),
+                tpu_udp_config.clone(),
+                tpu_quic_config.clone(),
             )
             .unwrap();
         let tpu_quic: Vec<UdpSocket> =
-            bind_more_with_config(tpu_quic, DEFAULT_QUIC_ENDPOINTS, quic_config.clone()).unwrap();
+            bind_more_with_config(tpu_quic, DEFAULT_QUIC_ENDPOINTS, tpu_quic_config.clone())
+                .unwrap();
+
+        let tpu_forwards_udp_config = SocketConfig::default();
+        let tpu_forwards_quic_config = SocketConfig::default().reuseport(true);
         let ((tpu_forwards_port, tpu_forwards), (_tpu_forwards_quic_port, tpu_forwards_quic)) =
             bind_two_in_range_with_offset_and_config(
                 bind_ip_addr,
                 port_range,
                 QUIC_PORT_OFFSET,
-                udp_config,
-                quic_config.clone(),
+                tpu_forwards_udp_config,
+                tpu_forwards_quic_config.clone(),
             )
             .unwrap();
         let tpu_forwards_quic = bind_more_with_config(
             tpu_forwards_quic,
             DEFAULT_QUIC_ENDPOINTS,
-            quic_config.clone(),
+            tpu_forwards_quic_config.clone(),
         )
         .unwrap();
-        let (tpu_vote_port, tpu_vote) = Self::bind(bind_ip_addr, port_range);
-        let (tpu_vote_quic_port, tpu_vote_quic) = Self::bind(bind_ip_addr, port_range);
+
+        let tpu_vote_udp_config = SocketConfig::default();
+        let tpu_vote_quic_config = SocketConfig::default().reuseport(true);
+        let (tpu_vote_port, tpu_vote) =
+            Self::bind_with_config(bind_ip_addr, port_range, tpu_vote_udp_config.clone());
+        // using udp port for quic really because we need to reusport set to false, since Self::bind() defaults to false
```

Comment:
```suggestion
        // using udp port for quic really because we need to set reusport to false, since Self::bind() defaults to false
```
nit: grammar

Also, can provide a little more detail on this piece; seems a little tricky

Reply:
honestly, no idea why i wrote this. doesn't really make sense. but happy to get some more details on this